Sources/Sentry/SentrySerialization.m

```diff
@@ -260,13 +269,28 @@ + (SentrySession *_Nullable)sessionWithData:(NSData *)sessionData
     return session;
 }
 
-+ (NSData *)dataWithReplayRecording:(SentryReplayRecording *)replayRecording
++ (NSData *_Nullable)dataWithReplayRecording:(SentryReplayRecording *)replayRecording
```

Comment:
I am not 100% certain if an early return approach makes sense here, or if we should return empty data instead.

I decided to go for the early return, as it does not make sense to me to return data e.g. without the header.

WDYT?

Reply:
In this case, I think an envelope item without a header is invalid, and we should do an early return. So please keep it as it is.